addons/l10n_in/models/res_partner.py

```diff
@@ -100,3 +100,5 @@ def action_update_state_as_per_gstin(self):
         self.ensure_one()
         state_id = self.env['res.country.state'].search([('l10n_in_tin', '=', self.vat[:2])], limit=1)
         self.state_id = state_id
```

Comment:
I wonder if we should not have put it on the write on the partner instead of the company.  (or on the inverse or something)

Reply:
Let's stick to this in stable.